### PR TITLE
Implem exo vm reset (restore)

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
Signed-off-by: Pierre-Emmanuel Jacquier <pierre-emmanuel.jacquier@epitech.eu>

### CODE REVIEW

**Comment by greut** [comment]
I guess, you'll have to send at least the `rootdisksize` option. @marcaurele do you confirm?

**Comment by greut** [comment]
like for the deletion, I'd ask the user confirmation before restoring everything from scratch :stuck_out_tongue_winking_eye:

**Comment by marcaurele** [comment]
Yes it's kind of mandatory to send the `rootdisksize` otherwise the template size will be used, which is now not the correct value due to the single template size at 10GB. For the reset operation the user can downscale its disk. It would be nice to present the current disk size and let it choose any other value in the usual boundaries: starting at the template size of the one selected, maxed by the SO of the VM.

File: cmd/exo/cmd/vm_reset.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+			}
+			return errors.New(b.String())
+		}
+
+		return nil
+	},
+}
+
+// resetVirtualMachine stop a virtual machine instance
+func resetVirtualMachine(vmName string, diskValue int64PtrValue, force bool) error {
```
Comment (greut) [inline cmd/exo/cmd/vm_reset.go]: you should reapply the current disk size.

```diff
- func resetVirtualMachine(vmName string, diskValue int64PtrValue, force bool) error {
+ func resetVirtualMachine(vmName string, force bool) error {
```

File: request.go [hunk 2]
```diff
@@ -361,11 +361,14 @@ func (client *Client) request(ctx context.Context, command Command) (json.RawMes
...
+	// XXX: addIpToNic, activateIp6, restorevmresponse are kind of special
+	switch key {
+	case "addiptonicresponse":
 		key = "addiptovmnicresponse"
-	} else if key == "activateip6response" {
+	case "activateip6response":
 		key = "activateip6nicresponse"
+	case "restorevirtualmachineresponse":
+		key = "restorevmresponse"
 	}
```
Comment (greut) [inline request.go]: maybe, we could have a map or but that into the client, like APIName/APIDescription

File: cmd/exo/cmd/vm_reset.go (hunk 1 above)
Comment (greut) [reply cmd/exo/cmd/vm_reset.go]: my bad, you did well.

File: request.go (hunk 2 above)
Comment (pierre-emmanuelJ) [reply request.go]: 👍

File: cmd/exo/cmd/vm_reset.go [hunk 3]
```diff
@@ -74,11 +74,21 @@ func resetVirtualMachine(vmName string, diskValue int64PtrValue, force bool) err
 		return err
 	}
 
-	rootDiskSize := int64(volume.Size >> 30)
+	volumeSize := volume.Size >> 30
+
+	temp := &egoscale.Template{IsFeatured: true, ID: vm.TemplateID, ZoneID: "1"}
```
Comment (greut) [inline cmd/exo/cmd/vm_reset.go]: reuse an existing ZoneID (cf #271)

File: cmd/exo/cmd/vm_reset.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		Type:             "ROOT",
+	}
+
+	if err := cs.Get(volume); err != nil {
+		return err
+	}
+
+	volumeSize := volume.Size >> 30
+
+	temp := &egoscale.Template{IsFeatured: true, ID: vm.TemplateID, ZoneID: "1128bd56-b4d9-4ac6-a7b9-c715b187ce11"}
```
Comment (greut) [inline cmd/exo/cmd/vm_reset.go]: `vm.ZoneID` ?

File: request.go [hunk 5]
```diff
@@ -361,11 +361,10 @@ func (client *Client) request(ctx context.Context, command Command) (json.RawMes
 
 	apiName := client.APIName(command)
 	key := fmt.Sprintf("%sresponse", strings.ToLower(apiName))
-	// XXX: addIpToNic, activateIp6 are kind of special
-	if key == "addiptonicresponse" {
-		key = "addiptovmnicresponse"
-	} else if key == "activateip6response" {
-		key = "activateip6nicresponse"
+	// XXX: addIpToNic, activateIp6, restorevmresponse are kind of special
```
Comment (greut) [inline request.go]: feel free to move the comment closer to the map.

File: cmd/exo/cmd/vm_reset.go (hunk 4 above)
Comment (pierre-emmanuelJ) [reply cmd/exo/cmd/vm_reset.go]: yes

**Pull request merged**